spatialos/UnrealGDK
# Bugfix/unr 1698 fix host url override

**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

**[UNR-1698] Correct receptionistHost command line functionality.**
The receptionistHost command line argument can now be used to override the default URL used by PIE clients. This only occurs on the first connection so subsequent connections will not use the receptionistHost URL override. This allows the PIE client to connect to a remote server on start up. Additionally connections using a LocatorConfig can now switch to a ReceptionistConfig. This allows you to connect to a local server if you were previously connected to the cloud. 

**Testing PIE client:**
Running a PIE client and adding a "receptionistHost" argument with a dodgy IP results in failure to connect. This is due to the "receptionistHost" argument overriding the default  IP used by the PIE client. This will allow a PIE client to initially connect to any server.

**Testing local client:**
Adding the "receptionistHost" argument does not affect running a "local" client. 

**Testing a deployment client:**
The deployment client after connecting to the cloud deployment can now connect to a local server.  

**QA:**
- Run a deployment. Also run a local server. Connect to a deployment server using a deployment client. Using the in game console, connect to the local server using the command: "Open 127.0.0.1".  You should now be connected to the server.

- Run a PIE client and server adding the command line argument: "+receptionistHost 1237.0.30.1". The client should fail to connect as the IP address is janky. Using the in game console, connect to the local server using the command: "Open 127.0.0.1". You should now be connected to the server.

Changes will be made to https://docs.improbable.io/unreal/alpha/content/map-travel.
Documentation PR: https://github.com/spatialos/UnrealGDK/pull/1431

@joshuahuburn 
@improbable-valentyn 

[UNR-1698]: https://improbableio.atlassian.net/browse/UNR-1698
